File: Android/Lab2/app/src/main/java/pl/szarek/activities/MainActivity.java
package pl.szarek.activities;

import android.content.Intent;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    private Button firstActivityButton;
    private Button secondActivityButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViews();
        setListeners();
    }

    private void findViews() {
        firstActivityButton = findViewById(R.id.first_activity_button);
        secondActivityButton = findViewById(R.id.second_activity_button);
    }

    private void setListeners() {
        firstActivityButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToFirstActivity();
            }
        });

        secondActivityButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                goToSecondActivity();
            }
        });
    }

    private void goToFirstActivity() {
        Intent intent = new Intent(this, FirstActivity.class);
        intent.putExtra("key_int", 123);
        startActivity(intent);
    }

    private void goToSecondActivity() {
        Intent intent = new Intent(this, SecondActivity.class);
        intent.putExtra("key_string", "Test string");
        startActivity(intent);
    }
}

File: aplikacje-mobilne-szarek-175ic/Lab4/app/src/main/java/pl/szarek/lab4/MainActivity.kt
package pl.szarek.lab4

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.RecyclerView
import pl.szarek.lab4.R

class MainActivity : AppCompatActivity() {
    private lateinit var typesRecyclerView: RecyclerView
    private lateinit var toolbar: Toolbar

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViews()
        initializeRecyclerView()
        setSupportActionBar(toolbar)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu_main_activity, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.info_item -> {
                goToInfoActivity()
                true
            }

            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun findViews() {
        toolbar = findViewById(R.id.toolbar)
        typesRecyclerView = findViewById(R.id.types_recycler_view)
    }

    private fun initializeRecyclerView() {
        val types = resources.getStringArray(R.array.types_array)
        val listener = object : RecyclerViewAdapter.ItemClickListener {
            override fun onClick(item: String) {
                goToCategoriesActivity()
            }
        }

        typesRecyclerView.adapter = RecyclerViewAdapter(types.toList(), listener)
    }

    private fun goToCategoriesActivity() {
        val intent = Intent(this, CategoriesActivity::class.java)
        startActivity(intent)
    }

    private fun goToInfoActivity() {
        val intent = Intent(this, InfoActivity::class.java)
        startActivity(intent)
    }
}
